network/chunks_download_strategy.hpp

```diff
@@ -32,20 +38,20 @@ class ChunksDownloadStrategy
 
   struct LessChunks
   {
-    bool operator() (ChunkT const & r1, ChunkT const & r2) const { return r1.m_pos < r2.m_pos; }
-    bool operator() (ChunkT const & r1, int64_t const & r2) const { return r1.m_pos < r2; }
-    bool operator() (int64_t const & r1, ChunkT const & r2) const { return r1 < r2.m_pos; }
+    bool operator()(ChunkT const & r1, ChunkT const & r2) const { return r1.m_pos < r2.m_pos; }
+    bool operator()(ChunkT const & r1, int64_t const & r2) const { return r1.m_pos < r2; }
+    bool operator()(int64_t const & r1, ChunkT const & r2) const { return r1 < r2.m_pos; }
   };
 
   using RangeT = std::pair<int64_t, int64_t>;
 
-  static const int SERVER_READY = -1;
+  static constexpr int SERVER_READY = -1;
   struct ServerT
   {
     std::string m_url;
     int m_chunkIndex;
 
-    ServerT(std::string const & url, int ind) : m_url(url), m_chunkIndex(ind) {}
+    ServerT(std::string  url, int ind) : m_url(std::move(url)), m_chunkIndex(ind) {}
```

Comment:
No &&?
```suggestion
    ServerT(std::string url, int ind) : m_url(std::move(url)), m_chunkIndex(ind) {}
```

Reply:
This is a clang-tidy's recommendation
https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html

Reply:
The link explicitly says that this approach (pass an argument by value) is beneficial when the function is used in two different ways: accepts a copy of the argument, or accepts a moveable object (e.g. RVO). Is that the case here? E.g. if the argument is always copied, or is always moved, then passing it explicitly as a const ref or && avoids unnecessary pessimisation.